nvtabular/ops/categorify.py

```diff
@@ -775,15 +836,41 @@ def _encode(
         if list_col:
             codes = cudf.DataFrame({selection_l[0]: gdf[selection_l[0]].list.leaves})
             codes["order"] = cp.arange(len(codes))
+            if buckets and selection_l[0] in buckets:
+                na_sentinel = _hash_bucket(gdf, buckets, selection_l[0])
         else:
-            codes = cudf.DataFrame({"order": cp.arange(len(gdf))})
+            codes = cudf.DataFrame({"order": cp.arange(len(gdf))}, index=gdf.index)
             for c in selection_l:
                 codes[c] = gdf[c].copy()
-        labels = codes.merge(
-            value, left_on=selection_l, right_on=selection_r, how="left"
-        ).sort_values("order")["labels"]
-        labels.fillna(na_sentinel, inplace=True)
-        labels = labels.values
+                if buckets and c in buckets and encode_type == "joint":
+                    na_sentinel = _hash_bucket(gdf, buckets, c)
+                elif buckets and encode_type == "combo":
+                    if len(selection_l) > 1:
+                        na_sentinel = _hash_bucket(
+                            gdf, buckets, tuple(selection_l), encode_type="combo"
+                        )
+                    else:
+                        na_sentinel = _hash_bucket(gdf, buckets, selection_l, encode_type="combo")
```

Comment:
@benfred if I do this change , it breaks `_hash_bucket` function calculation, so it would not work. 

1. selection_l is an array like ['author'] that's why I send `c` which is `author` to hash_bucket for `joint` encode type.
2. tuple(selection_l) is also required check based on how combo is designed. 

I am working to find a way to incorporate your change. Let's see.